src/contracts/interfaces/IPaymentCoordinator.sol

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./IStrategy.sol";
+import "./IPauserRegistry.sol";
+
+/**
+ * @title Interface for the `IPaymentCoordinator` contract.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @notice Allows AVSs to make "Range Payments", which get distributed amongst the AVSs' confirmed
+ * Operators and the Stakers delegated to those Operators.
+ * Calculations are performed based on the completed Range Payments, with the results posted in
+ * a Merkle root against which Stakers & Operators can make claims.
+ */
+interface IPaymentCoordinator {
+    /// STRUCTS ///
+    struct StrategyAndMultiplier {
+        IStrategy strategy;
+        // weight used to compare shares in multiple strategies against one another
+        uint96 multiplier;
+    }
+
+    struct RangePayment {
+        // Strategies & relative weights of shares in the strategies
+        StrategyAndMultiplier[] strategiesAndMultipliers;
+        IERC20 token;
+        uint256 amount;
+        uint64 startTimestamp;
+        uint64 duration;
+    }
+
+    struct DistributionRoot {
+        // merkle root of the distribution
+        bytes32 root;
+        // The timestamp until which payments have been calculated
+        uint64 paymentCalculationEndTimestamp;
+        // timestamp at which the root can be claimed against
+        uint64 activatedAt;
+    }
+
+    struct EarnerTreeMerkleLeaf {
+        address earner;
+        bytes32 earnerTokenRoot;
+    }
+
+    struct TokenTreeMerkleLeaf {
+        IERC20 token;
+        uint256 cumulativeEarnings;
+    }
+
+    struct PaymentMerkleClaim {
+        // The index of the root in the list of roots
+        uint32 rootIndex;
```

Comment:
can we add a view function to get the rootIndex from the root for the backend? could be inefficient reverse looping like middleware contracts

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/37b0406649f4056b78253278d75b49aef8d3a290/src/contracts/core/PaymentCoordinator.sol#L465

Added here